### PR TITLE
Chore/table header simplification

### PR DESCRIPTION
**Pull request by saltcod**
- tighten up header area
- implement intermediate state in select all button
- hide grid header actions when selections in place
 
screenshot-2024-11-01-at-16 53 32

### CODE REVIEW

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview

File: apps/studio/components/grid/components/footer/Footer.tsx [hunk 1]
```diff
@@ -20,6 +21,9 @@ const Footer = ({ isRefetching }: FooterProps) => {
   const { id: _id } = useParams()
   const id = _id ? Number(_id) : undefined
 
+  const state = useTrackedState()
+  const { selectedRows, allRowsSelected } = state
```
Comment (joshenlim) [inline apps/studio/components/grid/components/footer/Footer.tsx:25]: this seems unused - i'll remove it

File: apps/studio/components/grid/components/grid/SelectColumn.tsx [hunk 2]
```diff
@@ -183,13 +183,27 @@ function SelectCellHeader({
...
+  const state = useTrackedState()
+  const { selectedRows, allRowsSelected } = state
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  // indeterminate state === some rows are selected but not all
+  const isIndeterminate = selectedRows.size > 0 && !allRowsSelected
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.indeterminate = isIndeterminate
```
Comment (joshenlim) [inline apps/studio/components/grid/components/grid/SelectColumn.tsx:195]: hmm what does this do? are we using inputRef.indeterminate?

Comment (joshenlim) [reply apps/studio/components/grid/components/grid/SelectColumn.tsx:195]: will remove - seems like its unused

File: apps/studio/components/grid/components/header/Header.tsx [hunk 3]
```diff
@@ -377,23 +380,24 @@ const RowHeader = ({ table, sorts, filters }: RowHeaderProps) => {
...
-        </span>
-        {!allRowsSelected && totalRows > allRows.length && (
-          <Button type="link" onClick={() => onSelectAllRows()}>
-            Select all rows in table
-          </Button>
-        )}
-      </div>
-      <div className="h-[20px] border-r border-strong" />
       <div className="flex items-center gap-2">
+        <Tooltip_Shadcn_>
```
Comment (joshenlim) [inline apps/studio/components/grid/components/header/Header.tsx:384]: hmm the order of the buttons feel weird it terms of context
- we've got clear selection (which is to do with row selection)
- followed by export and delete
- then select all and the number of rows (which is to do with row selection)

image

Comment (joshenlim) [reply apps/studio/components/grid/components/header/Header.tsx:384]: any reason why we're changing from the current order of buttons?
image

File: apps/studio/components/grid/components/header/Header.tsx [hunk 4]
```diff
@@ -420,7 +424,7 @@ const RowHeader = ({ table, sorts, filters }: RowHeaderProps) => {
               <Button
                 type="default"
                 size="tiny"
-                icon={<Trash />}
+                icon={<Trash className="text-destructive-600" />}
```
Comment (joshenlim) [inline apps/studio/components/grid/components/header/Header.tsx:427]: hmm from a consistency POV: is this something we want to keep doing as a UI pattern?

**Review by joshenlim — APPROVED** [review]
LGTM! Tested header changes on the preview
- with at least one row selected but not all, header checkbox has a "-" icon
- Export to CSV + SQL works for selected rows (basic testing, didn't stress test this)

**Pull request merged**